engine/tests/phpunit/Elgg/Amd/ConfigTest.php

```diff
@@ -3,25 +3,37 @@
 
 
 class ConfigTest extends \PHPUnit_Framework_TestCase {
+	/**
+	 * @var \Elgg\PluginHooksService
+	 */
+	protected $hooks;
+
+	/**
+	 * @var \Elgg\Amd\Config
+	 */
+	protected $amdConfig;
+	
+	function setUp() {
+		$this->hooks = new \Elgg\PluginHooksService();
+		$amdConfig = new \Elgg\Amd\Config($this->hooks);
```

Comment:
I would discourage this refactoring. I know it feels like reducing duplication but it is much harder to reason about tests when you can't see the entire setup within the test method.

Instead, I might recommend a createAmdConfig method that returns the config instance instead of saving it to a property of the  testcase.

I also suspect this is broken (`$amdConfig` probably was meant to be `$this->amdConfig`)